scripts/generate_gitlab.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+
+import requests
+import yaml
+
+try:
+    gitlab_token = os.environ["GITLAB_TOKEN"]
+except KeyError:
+    print("Please provide the environment variable GITLAB_TOKEN")
+    sys.exit(1)
+
+parser = argparse.ArgumentParser(
+    description="Script to generate vcsconfig for all repositories \
+    under the given namespace (needs Gitlab >= 10.3)"
+)
+parser.add_argument("gitlab_host", type=str, help="url to the gitlab instance")
+parser.add_argument(
+    "gitlab_namespace",
+    type=str,
+    help="namespace/group in gitlab to generate vcsconfig for",
+)
+parser.add_argument(
+    "-c",
+    type=str,
+    help="path to the target config file (default: ./vcspull.yaml)",
+    dest="config_file_name",
+    required=False,
+    default="./vcspull.yaml",
+)
+
+args = vars(parser.parse_args())
+gitlab_host = args["gitlab_host"]
+gitlab_namespace = args["gitlab_namespace"]
+config_filename = args["config_file_name"]
+
+try:
+    if os.path.isfile(config_filename):
+        result = input(
+            "The target config file (%s) already exists, \
```

Comment:
As this codebase is for python >= 3.7, you can just use f-strings [PEP0498](https://www.python.org/dev/peps/pep-0498/). E.g. `f"foo {my_variable}"`

Reply:
@tony how do you stand on the topic of injecting variables into strings? is there one formatting / style you would like to see used in the python code?